src/main/kotlin/io/snyk/plugin/snykcode/core/SnykCodeUtils.kt

```diff
@@ -21,27 +21,29 @@ class SnykCodeUtils private constructor() : DeepCodeUtilsBase(
     fun allProjectFilesCount(project: Project): Int = allProjectFiles(project).size
 
     override fun allProjectFiles(projectO: Any): Collection<Any> {
+        scLogger.logInfo("allProjectFiles requested")
         val project = PDU.toProject(projectO)
-        return RunUtils.computeInReadActionInSmartMode(
+        val progressIndicator = ProgressManager.getInstance().progressIndicator ?: null
+        val allVirtualFiles = mutableSetOf<VirtualFile>()
+
+        val cancelled = !ProjectRootManager.getInstance(project).fileIndex.iterateContent {
+            if (!it.isDirectory) {
+                allVirtualFiles.add(it)
+            }
+            return@iterateContent progressIndicator?.isCanceled != true
```

Comment:
should we have another log entry here, if we return before :34 ?

Reply:
@michelkaporin Here we not return from outer fun but inner iterator ContentIterator.processFile() fun: "Returns: false if files processing should be stopped, true if it should be continued" I.e we just stop iterator in case user cancel task before it actually finished.
We'll pass through logging at :38 in any case.

Reply:
@ArtsiomCh got it, thanks!